test/tests/functional/pbs_modifyresv_hook.py

```diff
@@ -0,0 +1,284 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2020 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of both the OpenPBS software ("OpenPBS")
+# and the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# OpenPBS is free software. You can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version.
+#
+# OpenPBS is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public
+# License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# PBS Pro is commercially licensed software that shares a common core with
+# the OpenPBS software.  For a copy of the commercial license terms and
+# conditions, go to: (http://www.pbspro.com/agreement.html) or contact the
+# Altair Legal Department.
+#
+# Altair's dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of OpenPBS and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair's trademarks, including but not limited to "PBS™",
+# "OpenPBS®", "PBS Professional®", and "PBS Pro™" and Altair's logos is
+# subject to Altair's trademark licensing policies.
+
+import textwrap
+from tests.functional import *
+
+
+class TestModifyResvHook(TestFunctional):
+    """
+    Tests to verify the reservation begin hook for a confirm standing/advance/
+    degraded reservation once the reservation begins.
+    """
+
+    advance_resv_hook_script = textwrap.dedent("""\
+        import pbs
+        e=pbs.event()
+
+        pbs.logmsg(pbs.LOG_DEBUG, 'Reservation Modify Hook name - %s' %
+                   e.hook_name)
+
+        if e.type == pbs.MODIFYRESV:
+            pbs.logmsg(pbs.LOG_DEBUG, 'Reservation ID - %s' % e.resv.resvid)
+    """)
+
+    def setUp(self):
+        """
+        Create a reservation begin hook and set the server log level.
+        """
+        super(TestModifyResvHook, self).setUp()
```

Comment:
So, I just checked this using the tests in pbs_resv_begin_hook.py to see what kind of effect this has.  I was getting an average overall runtime of 908.729 sec when using super() vs 922.208 sec (this was a simple test using time to get the overall runtime user and sys times were also faster).  Also, it's pretty clear from python 3's documentation that super() is the preferred way to get at a parent call like this.  In terms of performance, I believe that the difference here is going to get swamped by noise and jitter on the host executing the tests.

This could be changed to the python3-only super() syntax and drop the TestModifyResvHook and self arguments, but the python2 style works (these are effectively implicit in the python3 version here).

Also this was the same style used in the existing resv_end tests.  Overall I think it's a better practice to use super() here.

Reply:
Wow, I wouldn't have thought super() would be faster than a direct function call.  I'll bring this up to the PTL maintainers for a change in guidance in how PTL tests should be written.  Thank you for doing the testing.

Reply:
The end result should be to leave super() in at this point.  I believe this conversation is resolved?